shap/utils/_legacy.py

```diff
@@ -96,9 +98,18 @@ def __init__(self, f, out_names):
 
 def convert_to_model(val):
     if isinstance(val, Model):
-        return val
+        out = val
     else:
-        return Model(val, None)
+        out = Model(val, None)
+
+    # Fix for the sklearn warning
+    # 'X does not have valid feature names, but <model> was fitted with feature names'
+    out = copy.deepcopy(out)
```

Comment:
I don't think we need to copy all the time. How about shifting this into the if-block?